src/convert.cpp

```diff
@@ -61,49 +73,60 @@ ddwaf_object* to_ddwaf_object_object(
   Napi::Object obj,
   int depth,
   bool lim,
-  bool ignoreToJSON
+  bool ignoreToJSON,
+  JsSet jsSet
 ) {
   if (!ignoreToJSON) {
     Napi::Value toJSON = obj.Get("toJSON");
     if (toJSON.IsFunction()) {
-      return to_ddwaf_object(object, env, toJSON.As<Napi::Function>().Call(obj, {}), depth, lim, true);
+      return to_ddwaf_object(object, env, toJSON.As<Napi::Function>().Call(obj, {}), depth, lim, true, jsSet);
     }
   }
 
+  if (jsSet.Has(obj)) {
+    mlog("Circular dependency")
+    return ddwaf_object_invalid(object);
+  }
+  jsSet.Add(obj);
```

Comment:
instead of adding it here, and deleting it when possible failures happen, why not just adding the obj after the possible failures, like line 110 or so

Reply:
not here anymore, it is now in main `to_ddwaf_object()` method